src/storage/mvcc/reader/point_getter.rs

```diff
@@ -0,0 +1,1656 @@
+    }
+
+    /// Build `PointGetter` from the current configuration.
+    pub fn build(self) -> Result<PointGetter<S>> {
+        let write_cursor = CursorBuilder::new(&self.snapshot, CF_WRITE)
+            .fill_cache(self.fill_cache)
+            .prefix_seek(!self.multi)
+            .build()?;
+
+        Ok(PointGetter {
+            snapshot: self.snapshot,
+            multi: self.multi,
+            fill_cache: self.fill_cache,
+            omit_value: self.omit_value,
+            isolation_level: self.isolation_level,
+
+            statistics: Statistics::default(),
+
+            read_once: false,
+
+            write_cursor,
+            default_cursor: None,
+        })
+    }
+}
+
+/// This struct can be used to get the value of a user key. Internally, rollbacks are ignored and
+/// smaller version will be tried. If the isolation level is SI, locks will be checked first.
+///
+/// If `multi` is `false`, prefix filter will be used so that you can only call `read_next`
+/// once. Subsequent calls will yield `None`.
+///
+/// If `multi` is `true`, the instance can be re-used to get multiple keys. However it will
+/// be optimal if these keys are get in ascending order and are relatively close to each other.
+///
+/// Use `PointGetterBuilder` to build `PointGetter`.
+pub struct PointGetter<S: Snapshot> {
+    snapshot: S,
+    multi: bool,
+    fill_cache: bool,
+    omit_value: bool,
+    isolation_level: IsolationLevel,
+
+    statistics: Statistics,
+
+    /// Whether there is already a `read_next` call. When `multi == false`, we use this field
+    /// to check that `read_next` is called only once.
+    read_once: bool,
+
+    write_cursor: Cursor<S::Iter>,
+
+    /// Default cursor is optional since when value is short we don't need to look up in
+    /// the default CF.
+    default_cursor: Option<Cursor<S::Iter>>,
+}
+
+impl<S: Snapshot> PointGetter<S> {
+    /// Take out and reset the statistics collected so far.
+    #[inline]
+    pub fn take_statistics(&mut self) -> Statistics {
+        ::std::mem::replace(&mut self.statistics, Statistics::default())
+    }
+
+    /// Get the value of a user key. See `PointGetter` for details.
+    pub fn read_next(&mut self, user_key: &Key, mut ts: u64) -> Result<Option<Value>> {
+        // Attempting to read multiple values when `multi == false`, directly returns `None`.
+        if !self.multi && self.read_once {
+            return Ok(None);
+        }
+
+        self.read_once = true;
+
+        if self.isolation_level == IsolationLevel::SI {
+            // Check for locks that signal concurrent writes in SI.
+            ts = super::util::load_and_check_lock(
+                &self.snapshot,
+                user_key,
+                ts,
+                &mut self.statistics,
+            )?;
+        }
+
+        // First seek to `${user_key}_${ts}`. In multi-read mode, the keys may given out of
+        // order, so we allow re-seek.
+        self.write_cursor.near_seek(
```

Comment:
Why not use seek directly? For PointGetter is there any case we can benefit by near_seek?

Reply:
When we get multiple keys and these keys are sorted in ascending order, near seek will be faster.